ansible/roles/vm_set/tasks/kickstart_vm.yml

```diff
@@ -27,43 +27,19 @@
     delay: 10
     ignore_errors: true
 
-  - name: Destroy vm {{ vm_name }} if it hangs
-    virt: name={{ vm_name }}
-          command=destroy
-          uri=qemu:///system
-    when: kickstart_output.kickstart_code != 0
-    become: yes
-    ignore_errors: true
-
-  - name: Start vm again {{ vm_name }}
-    virt: name={{ vm_name }}
-          state=running
-          uri=qemu:///system
-    when: kickstart_output.kickstart_code != 0
-    become: yes
-    ignore_errors: true
-
-  - name: Wait until vm {{ vm_name }} is loaded
-    kickstart: telnet_port={{ serial_port }}
-              login={{ eos_default_login }}
-              password={{ eos_default_password }}
-              hostname={{ hostname }}
-              mgmt_ip="{{ mgmt_ip_address }}/{{ mgmt_prefixlen }}"
-              mgmt_gw={{ vm_mgmt_gw | default(mgmt_gw) }}
-              new_login={{ eos_login }}
-              new_password={{ eos_password }}
-              new_root_password={{ eos_root_password }}
-    register: kickstart_output_final
-    until: '"kickstart_code" in kickstart_output_final and kickstart_output_final.kickstart_code == 0'
-    retries: 5
-    delay: 10
+  - name: Respin failed vm
+    include_tasks: respin_vm.yml
+    vars:
+      src_disk_image: "{{ home_path }}/{{ root_path }}/images/{{ hdd_image_filename }}"
+      disk_image: "{{ home_path }}/{{ root_path }}/disks/{{ vm_name }}_hdd.vmdk"
+      cdrom_image: "{{ home_path }}/{{ root_path }}/images/{{ cd_image_filename }}"
+    when: '"kickstart_code" in kickstart_output and kickstart_output.kickstart_code != 0'
     ignore_errors: true
-    when: kickstart_output.kickstart_code != 0
 
-  - name: Kickstart gives error again vm {{ vm_name }}
+  - name: Kickstart gives error after respin vm {{ vm_name }}
     set_fact:
       kickstart_failed_vms: "{{ kickstart_failed_vms + [vm_name] }}"
-    when: '"kickstart_code" in kickstart_output_final and kickstart_output_final.kickstart_code != 0'
+    when: '"kickstart_code" in kickstart_output and kickstart_output.kickstart_code != 0'
```

Comment:
Is it possible to retry respine if one round of respine still failed? To avoid endless retry, we can add a max retry limitation, like 3 times?

Reply:
Good suggestion. I'll make this change in next PR.